### ISSUE TITLE
Random split does not work correctly

### ISSUE THREAD

**Issue by Myname10** [lyst/lightfm#599]
Hello,

I'm using the random split tain test funtion with a fixed seed  

train, test = random_train_test_split(interactions=lfm_interactions_with_weights,
                                              test_percentage=test_percentage, random_state=random.RandomState(3))

But whenver i run this function it returrns different train, test 

ANy help thank you
